src/main/scala/org/broadinstitute/hail/asm4s/Code.scala

```diff
@@ -238,7 +392,8 @@ object Invokeable {
         parameterTypes.map(_.getName).mkString(", ")
       })")
 
-    assert(m.getReturnType == sct.runtimeClass,
+    // generic type parameters return java.lang.Object instead of the correct class
+    assert(m.getReturnType.isAssignableFrom(sct.runtimeClass),
```

Comment:
I need to think more about this. Generic parameters complicate things. I think there should be a `Code.checkcast[S]` in here.